editor/import/editor_import_plugin.cpp

```diff
@@ -0,0 +1,152 @@
+/*************************************************************************/
+#include "editor_import_plugin.h"
+#include "core/script_language.h"
+
+EditorImportPlugin::EditorImportPlugin() {
+}
+
+String EditorImportPlugin::get_importer_name() const {
+	ERR_FAIL_COND_V(!(get_script_instance() && get_script_instance()->has_method("get_importer_name")), "");
+	return get_script_instance()->call("get_importer_name");
+}
+
+String EditorImportPlugin::get_visible_name() const {
+	ERR_FAIL_COND_V(!(get_script_instance() && get_script_instance()->has_method("get_visible_name")), "");
+	return get_script_instance()->call("get_visible_name");
+}
+
+void EditorImportPlugin::get_recognized_extensions(List<String> *p_extensions) const {
+	ERR_FAIL_COND(!(get_script_instance() && get_script_instance()->has_method("get_recognized_extensions")));
+	Array extensions = get_script_instance()->call("get_recognized_extensions");
+	for (int i = 0; i < extensions.size(); i++) {
+		p_extensions->push_back(extensions[i]);
+	}
+}
+
+String EditorImportPlugin::get_preset_name(int p_idx) const {
+	ERR_FAIL_COND_V(!(get_script_instance() && get_script_instance()->has_method("get_preset_name")), "");
+	return get_script_instance()->call("get_preset_name", p_idx);
+}
+
+int EditorImportPlugin::get_preset_count() {
+	ERR_FAIL_COND_V(!(get_script_instance() && get_script_instance()->has_method("get_preset_count")), 0);
+	return get_script_instance()->call("get_preset_count");
+}
+
+String EditorImportPlugin::get_save_extension() const {
+	ERR_FAIL_COND_V(!(get_script_instance() && get_script_instance()->has_method("get_save_extension")), "");
+	return get_script_instance()->call("get_save_extension");
+}
+
+String EditorImportPlugin::get_resource_type() const {
+	ERR_FAIL_COND_V(!(get_script_instance() && get_script_instance()->has_method("get_resource_type")), "");
+	return get_script_instance()->call("get_resource_type");
+}
+
+void EditorImportPlugin::get_import_options(List<ResourceImporter::ImportOption> *r_options, int p_preset) const {
+
+	ERR_FAIL_COND(!(get_script_instance() && get_script_instance()->has_method("get_import_options")));
+	Array needed;
+	needed.push_back("name");
+	needed.push_back("default_value");
+	Array options = get_script_instance()->call("get_import_options", p_preset);
+	for (int i = 0; i < options.size(); i++) {
+		Dictionary d = options[i];
+		ERR_FAIL_COND(!d.has_all(needed));
+		String name = d["name"];
+		Variant default_value = d["default_value"];
+
+		PropertyHint hint = PROPERTY_HINT_NONE;
+		if (d.has("property_hint")) {
+			hint = (PropertyHint)d["property_hint"].operator int64_t();
+		}
+
+		String hint_string;
+		if (d.has("hint_string")) {
+			hint_string = d["hint_string"];
+		}
+
+		uint32_t usage = PROPERTY_USAGE_DEFAULT;
+		if (d.has("usage")) {
+			usage = d["usage"];
+		}
+
+		ImportOption option(PropertyInfo(default_value.get_type(), name, hint, hint_string, usage), default_value);
+		r_options->push_back(option);
```

Comment:
Maybe we should make the `ImportOption` struct a resource class?
So it's easier to construct from scripts and you won't have to fill a dictionary..